CCDB/include/CCDB/BasicCCDBManager.h

```diff
@@ -311,6 +315,19 @@ T* CCDBManagerInstance::getForTimeStamp(std::string const& path, long timestamp)
   return ptr;
 }
 
+template <typename T>
+T* CCDBManagerInstance::getForRunMiddleTimeStamp(std::string const& path, int runNumber, bool setRunMetadata)
+{
+  auto [start, stop] = getRunDuration(runNumber);
+  if (start < 0 || stop < 0) {
+    if (mFatalWhenNull) {
+      reportFatal(std::string("Failed to get run duration for run ") + std::to_string(runNumber));
+    }
+    return nullptr;
+  }
+  return setRunMetadata ? getForTimeStamp<T>(path, start / 2 + stop / 2, {{"runNumber", std::to_string(runNumber)}}) : getForTimeStamp<T>(path, start / 2 + stop / 2);
```

Comment:
Thanks for fixing my swapped `if` condition. But `getForTimeStamp` does not accept metadata, one should use `getSpecific` instead.

Reply:
Ah, you are right, sorry! Looking inside `getSpecific` in the end we just need to set the `mMetaData` variable, have a look if it is correct!